loader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/loader/LoaderCache.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2018-2020 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.storage.bigquery
+package loader
+
+import org.joda.time.Instant
+
+import cats.data.NonEmptyList
+import com.snowplowanalytics.iglu.core.SchemaKey
+import com.snowplowanalytics.iglu.schemaddl.bigquery.Field
+import com.snowplowanalytics.snowplow.badrows.FailureDetails
+
+import scala.collection.mutable.{Map => MMap}
+
+/** Singleton used to cache the results of looking up a SchemaKey and transforming the
+  * resulting JSON into a [[Field]].
+  */
+object LoaderCache {
+  type Key = SchemaKey
+  type Value = Either[NonEmptyList[FailureDetails.LoaderIgluError], Field]
+  type Deadline = Long
+
+  private[loader] var underlying = MMap.empty[Key, (Value, Deadline)]
+  private def now: Long = Instant.now.getMillis
+
+  def get(key: Key): Option[(Value, Deadline)] = underlying.get(key)
+
+  // Overrides existing value for the same key. Sets ttl to 5 mins.
+  def put(key: Key, value: Value, ttl: Long = 5 * 60 * 1000): Value = {
```

Comment:
IMO it would be better to return `Unit` instead of returning an input parameter

Reply:
We're only using `put` when we cannot find the value in the cache. And if we've looked for it, then we need it... Do you agree?

Reply:
Deleted my comment, as I ended up agreeing that `Unit` is indeed better.